starknet-rpc-test/get_transaction_receipt.rs

```diff
@@ -265,3 +268,91 @@ async fn fail_invalid_transaction_hash(#[future] madara: MadaraClient) -> Result
 
     Ok(())
 }
+
+#[rstest]
+#[tokio::test]
+async fn work_with_messages_to_l1(#[future] madara: MadaraClient) -> Result<(), anyhow::Error> {
```

Comment:
just need to add tests for edge cases:
- What happens if the tx fail ?
- Is there any limit on the number of messages per tx ?

Reply:
> What happens if the tx fail ?

I don't get what you mean. If the tx fail there is not tx receipt